### PR TITLE
[ui] Show Agent activity on the timeline

### PR DESCRIPTION
**Pull request by htin1**
## Summary

- Show green glow/fade rings around clips the Agent adds and orange rings around clips it actually changes.
- Show orange track-header rings for Agent track reorder, mute/hide, and sync-lock changes.
- Show slate `#64748B` indicators for clip-scoped and explicit timeline-range reads so users can see what the Agent is inspecting.
- Suppress indicators for no-ops, metadata-only reads, disjoint/unwindowed reads, tool errors, active-timeline changes, and interleaved manual edits.

## Approach

- Classify successful Agent timeline mutations from before/after timeline values, with explicit mutation-tool eligibility and write-over-read precedence.
- Route content-reading tools into the same unified `AgentActivityHighlight` state used by writes; reads remain visible while work is active, then hold and fade.
- Render activity through shared masked Core Animation layers that stay below the sticky ruler and update independently from full timeline redraws.
- Keep activity transient and session-only: no project persistence, undo entries, selection changes, or Agent contract changes.
- Track removals and pure clip deletions intentionally remain unanimated because there is no surviving visual target.

## Testing

- `swift build` — passed after rebase.
- `swift test --filter 'AgentActivityHighlightTests|MCPCopyClipSettingsTests'` — 15 tests across 2 suites passed.
- `swift test` — 1,480 tests across 235 suites passed.
- 5,000-clip classification measured around 11 ms in a debug build; deterministic coverage verifies all 5,000 mutations are classified.
- Exercised add, mutate, copy settings, clip read, range read, reorder, hide, and sync-lock flows through the local MCP server in isolated test projects.
- Claude Opus 5 reviewed the complete branch for edge cases, bugs, concurrency, rendering, performance, and simplification; all blocking findings were addressed.

Manual UI verification still required:

- [ ] Confirm added, mutated, read, and track-header colors/glows in Light and Dark appearances.
- [ ] Scroll horizontally and vertically while an indicator is active.
- [ ] Zoom while an indicator is active.
- [ ] Collapse and reopen the timeline panel during an indicator.
- [ ] Confirm a write immediately followed by a read preserves the write indicator.
- [ ] Confirm reordered track rings follow the resulting rows.

## Change statistics

- Agent classification and activity state: +264 / -3
- Timeline rendering and theme: +310 / -0
- Tests: +256 / -1
- Total: +830 / -4

### CODE REVIEW

File: Sources/PalmierPro/Agent/Tools/ToolExecutor+AgentActivity.swift [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        .manageClipLinks,
+        .addClips, .insertClips, .moveClips, .removeClips,
+        .splitClips, .rippleDeleteRanges, .swapClipMedia,
+        .setClipProperties, .setKeyframes, .applyLayout, .syncClips, .undo,
+        .manageMulticam, .changeCam,
+        .removeWords, .removeSilence,
+        .addTexts, .updateText, .addCaptions,
+        .applyColor, .applyEffect, .denoiseAudio,
+        .generateAudio,
+    ]
```
[the remark this answers is not in the document]
Comment (htin1) [reply Sources/PalmierPro/Agent/Tools/ToolExecutor+AgentActivity.swift:19]: Fixed in feda90a0. copy_clip_settings now participates in mutation activity, and focused coverage verifies that only destination clips whose settings actually changed receive orange rings; the source and no-op repeats remain unhighlighted.
